embassy-executor/src/raw/waker.rs

```diff
@@ -46,7 +46,31 @@ pub fn task_from_waker(waker: &Waker) -> TaskRef {
     unsafe { TaskRef::from_ptr(hack.data as *const TaskHeader) }
 }
 
+#[cfg(not(feature = "nightly"))]
 struct WakerHack {
     data: *const (),
     vtable: &'static RawWakerVTable,
 }
+
+/// Get a task pointer from a waker.
```

Comment:
can you put the cfgs inside the function, so the function doc comment and the signature aren't duplicated, to ensure they don't go out of sync?